mavros_extras/src/plugins/odom.cpp

```diff
@@ -341,46 +232,30 @@ class OdometryPlugin : public plugin::PluginBase {
 		Matrix6d r_vel = Matrix6d::Zero();	//!< Zero initialized velocity 6-D Covariance matrix. WRT child_frame_id
 
 		mavlink::common::msg::ODOMETRY msg {};
+		msg.frame_id = utils::enum_value(MAV_FRAME::VISION_NED);
+		msg.child_frame_id = utils::enum_value(MAV_FRAME::BODY_FRD);
 
 		/**
-		 * Position and orientation are in the same frame as frame_id.
-		 * For a matter of simplicity, and given the existent MAV_FRAME
-		 * enum values, the default frame_id will be a local frame of
-		 * reference, so the pose is WRT a local frame.
+		 * Position parsing from odometry's parent frame to "vision_ned" frame.
 		 */
-		position = Eigen::Vector3d(tf_parent2local.linear() * ftf::to_eigen(odom->pose.pose.position));
-
-		// Orientation represented by a quaternion rotation from the local frame to XYZ body frame
-		Eigen::Quaterniond q_parent2child(ftf::to_eigen(odom->pose.pose.orientation));
-		Eigen::Affine3d tf_local2body = tf_parent2local * q_parent2child * tf_child2body.inverse();
-		orientation = Eigen::Quaterniond(tf_local2body.linear());
-
-		r_pose.block<3, 3>(0, 0) = r_pose.block<3, 3>(3, 3) = tf_parent2local.linear();
-
-		msg.frame_id = utils::enum_value(lf_id);
+		position = Eigen::Vector3d(tf_parent2parentDes.linear() * ftf::to_eigen(odom->pose.pose.position));
 
 		/**
-		 * Linear and angular velocities are in the same frame as child_frame_id.
-		 * Same logic here applies as above.
+		 * Orientation parsing.
 		 */
-		auto set_tf = [&](Eigen::Affine3d affineTf, MAV_FRAME frame_id) {
-				lin_vel = Eigen::Vector3d(affineTf.linear() * ftf::to_eigen(odom->twist.twist.linear));
-				ang_vel = Eigen::Vector3d(affineTf.linear() * ftf::to_eigen(odom->twist.twist.angular));
-				r_vel.block<3, 3>(0, 0) = r_vel.block<3, 3>(3, 3) = affineTf.linear();
-
-				msg.child_frame_id = utils::enum_value(frame_id);
-			};
+		Eigen::Quaterniond q_child2parent(ftf::to_eigen(odom->pose.pose.orientation));
+		Eigen::Affine3d tf_childDes2parentDes = tf_parent2parentDes * q_child2parent * tf_child2childDes.inverse();
+		orientation = Eigen::Quaterniond(tf_childDes2parentDes.linear());
 
-		if (odom->child_frame_id == "world" || odom->child_frame_id == "odom") {
-			// the child_frame_id would be the same reference frame as frame_id
-			set_tf(tf_child2local, lf_id);
-		}
-		else {
-			// the child_frame_id would be the WRT a body frame reference
-			set_tf(tf_child2body, bf_id);
-		}
+		/**
+		 * Linear and angular velocities has to be fcu_frd frame
```

Comment:
LOCAL_FRD?

Reply:
Velocities are expressed in body frame. So the right frame would be base_link_frd